src/features/counterfactual/utils.ts

```diff
@@ -356,27 +356,20 @@ export const extractCounterfactualSafeSetup = (
       saltNonce: string | undefined
     }
   | undefined => {
-  if (!undeployedSafe || !chainId) {
+  if (!undeployedSafe || !chainId || !undeployedSafe.props.safeAccountConfig) {
     return undefined
   }
-  if (isPredictedSafeProps(undeployedSafe.props)) {
-    return {
-      owners: undeployedSafe.props.safeAccountConfig.owners,
-      threshold: undeployedSafe.props.safeAccountConfig.threshold,
-      fallbackHandler: undeployedSafe.props.safeAccountConfig.fallbackHandler,
-      safeVersion: undeployedSafe.props.safeDeploymentConfig?.safeVersion,
-      saltNonce: undeployedSafe.props.safeDeploymentConfig?.saltNonce,
-    }
-  } else {
-    const { owners, threshold, fallbackHandler } = undeployedSafe.props.safeAccountConfig
-
-    return {
-      owners,
-      threshold: Number(threshold),
-      fallbackHandler,
-      safeVersion: undeployedSafe.props.safeVersion,
-      saltNonce: undeployedSafe.props.saltNonce,
-    }
+  const { owners, threshold, fallbackHandler } = undeployedSafe.props.safeAccountConfig
+  const { safeVersion, saltNonce } = isPredictedSafeProps(undeployedSafe.props)
+    ? undeployedSafe.props.safeDeploymentConfig ?? {}
+    : undeployedSafe.props
```

Comment:
I'm all for turning this on: https://eslint.org/docs/latest/rules/no-nested-ternary. Those ternaries always give me the chills.

Reply:
Let's agree that different people have different views on what is easier to read :)